cats/src/test/scala/sttp/tapir/codec/cats/TapirCodecCatsTest.scala

```diff
@@ -30,4 +36,64 @@ class TapirCodecCatsTest extends FlatSpec with Matchers {
 
     implicitly[Validator[NonEmptyChain[Test]]].show shouldBe expectedValidator.show
   }
+
+  implicit def arbitraryNonEmptyList[T: Arbitrary]: Arbitrary[NonEmptyList[T]] = Arbitrary(
+    Gen.nonEmptyListOf(implicitly[Arbitrary[T]].arbitrary).map(NonEmptyList.fromListUnsafe(_))
+  )
+
+  implicit def arbitraryNonEmptyChain[T: Arbitrary]: Arbitrary[NonEmptyChain[T]] = Arbitrary(
+    Gen.nonEmptyListOf(implicitly[Arbitrary[T]].arbitrary).map(NonEmptyChain.fromSeq(_).get)
+  )
+
+  implicit def arbitraryNonEmptySet[T: Arbitrary: Ordering]: Arbitrary[NonEmptySet[T]] = Arbitrary(
+    Gen.nonEmptyBuildableOf[SortedSet[T], T](implicitly[Arbitrary[T]].arbitrary).map(NonEmptySet.fromSet(_).get)
+  )
+
+  "Provided PlainText coder for non empty list" should "correctly serialize a non empty list" in {
```

Comment:
I know I could have factorize test logic. But when its not too much of a burden I like to keep test simple.

Reply:
looks good :)